polars/dataframe.ts

```diff
@@ -1707,6 +1704,55 @@ function map(df: DataFrame, fn: (...args: any[]) => any[]) {
   return df.rows().map(fn);
 }
 
+function mapPolarsTypeToJSONSchema(colType: DataType): string {
+  const typeMapping: { [key: string]: string } = {
+    Null: "null",
+    Bool: "boolean",
+    Int8: "integer",
+    Int16: "integer",
+    Int32: "integer",
+    Int64: "integer",
+    UInt8: "integer",
+    UInt16: "integer",
+    UInt32: "integer",
+    UInt64: "integer",
+    Float32: "number",
+    Float64: "number",
+    Date: "string",
+    Datetime: "string",
+    Utf8: "string",
+    Categorical: "string",
+    List: "array",
+    Struct: "object",
+  };
+
+  const dataType = colType.variant;
+  return typeMapping[dataType] || "string";
+}
+
+/////// HTML ESCAPING ///////
+const rawToEntityEntries = [
+  ["&", "&amp;"],
+  ["<", "&lt;"],
+  [">", "&gt;"],
+  ['"', "&quot;"],
+  ["'", "&#39;"],
+] as const;
+
+const rawToEntity = new Map<string, string>(rawToEntityEntries);
+
+const rawRe = new RegExp(`[${[...rawToEntity.keys()].join("")}]`, "g");
+
+/**
+ * Escapes text for safe interpolation into HTML text content and quoted attributes
+ */
+export function escapeHTML(str: string) {
+  return str.replaceAll(rawRe, (m) => {
+    const entity = rawToEntity.get(m);
+    return entity !== undefined ? entity : m;
```

Comment:
I think we can shorten this to: `return entity ?? m`

Reply:
Sweet. Originally I had this as a non-null assertion and then typescript complained about it with the current rules. This will work.